projects/sandbox/sampling/sample.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from pathlib import Path
+from time import time
+from typing import List
+
+import bilby
+import h5py
+import numpy as np
+import pandas as pd
+import torch
+
+from ml4gw.transforms import ChannelWiseScaler
+from mlpe.architectures import architecturize
+from mlpe.data.transforms import Preprocessor
+from mlpe.injection.priors import sg_uniform
+from mlpe.logging import configure_logging
+
+
+def _load_preprocessor_state(preprocessor, preprocessor_dir):
+    preprocessor_path = Path(preprocessor_dir) / "preprocessor"
+
+    whitener_path = preprocessor_path / "whitener.pt"
+    scaler_path = preprocessor_path / "scaler.pt"
+
+    preprocessor.whitener = torch.load(whitener_path)
+    preprocessor.scaler = torch.load(scaler_path)
+    return preprocessor
+
+
+def _load_test_data(testing_path: Path, inference_params: List[str]):
+    with h5py.File(testing_path, "r") as f:
+        signals = f["injections"][:]
+        params = []
+        for param in inference_params:
+            values = f["phase" if param == "phi" else param][:]
```

Comment:
There is an ambiguity / bug that we should address (which might explain why the phi variable looked so poorly in the PP plot you showed): in `ml4gw`, `phi` refers to `ra`, not the phase.